src/_pytest/main.py

```diff
@@ -418,10 +448,11 @@ def __init__(self, config: Config) -> None:
         self.config.pluginmanager.register(self, name="session")
 
     @classmethod
-    def from_config(cls, config):
-        return cls._create(config)
+    def from_config(cls, config: Config) -> "Session":
+        session = cls._create(config)  # type: Session
+        return session
```

Comment:
Better to annotate the `_create` method here than to override type inference.

Reply:
`_create()` is in a similar situation to `from_parent`, at least changing it would conflict with #6387, so I work around it too.

Reply:
https://github.com/pytest-dev/pytest/pull/6387 has been merged, so that might unblock this a bit.